vm/src/stdlib/builtins.rs

```diff
@@ -248,11 +248,37 @@ mod builtins {
     #[cfg(feature = "rustpython-compiler")]
     #[pyfunction]
     fn eval(
-        source: Either<PyStrRef, PyRef<crate::builtins::PyCode>>,
+        source: Either<ArgStrOrBytesLike, PyRef<crate::builtins::PyCode>>,
         scope: ScopeArgs,
         vm: &VirtualMachine,
     ) -> PyResult {
-        run_code(vm, source, scope, compile::Mode::Eval, "eval")
+        // source as string
+        let code = match source {
+            Either::A(either) => {
+                let source: &[u8] = &either.borrow_bytes();
+                for x in source {
+                    if *x == 0 {
+                        return Err(vm.new_value_error(
+                            "source code string cannot contain null bytes".to_owned(),
+                        ));
+                    }
+                }
```

Comment:
```suggestion
                if source.iter().any(|&b| b == 0) {
                    return Err(vm.new_value_error(
                        "source code string cannot contain null bytes".to_owned(),
                    ));
                }
```

Reply:
https://doc.rust-lang.org/nightly/core/iter/trait.Iterator.html#method.any

> any() is short-circuiting; in other words, it will stop processing as soon as it finds a true, given that no matter what else happens, the result will also be true.

It's simple but powerful 🚀